Add initial draft of Podman project Governance (#25398)
containers/podman: GOVERNANCE.md

    * Answer questions from other community members on the mailing list, Slack, Matrix, and other communication channels
    * Assist in triaging issues, following the [issue triage guide](./TRIAGE.md)
    * Assist in reviewing pull requests, including testing patches when applicable
    * Test release candidates and provide feedback
    * Promote the project in public
    * Help run the project infrastructure
* Privileges:
    * Invitations to contributor events
    * Eligible to become a Reviewer

### Reviewer
Description: A Reviewer has responsibility for the triage of issues and review of pull requests on the Podman project or a subproject, consisting of one or more of the Git repositories that form the project. They are collectively responsible, with other Reviewers, for reviewing changes to the repository or repositories and indicating whether those changes are ready to merge. They have a track record of contribution and review in the project.

Reviewers have all the rights and responsibilities of a Contributor, plus:

* Responsibilities include:
    * Regular contribution of pull requests to the Podman project or its subprojects

>     * Triage of Github issues on the Podman project or its subprojects

Reviewer: "Github" s/b "GitHub" throughout.

    * Following the reviewing guide
    * A sustained high level of pull request reviews on the Podman project or one of its subprojects
    * Assisting new Contributors with the process of contributing
    * Helping other contributors become reviewers
* Requirements:
    * Has a proven record of good-faith contributions to the project as a Contributor for a period of at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers.
    * Has participated in pull request review and/or issue triage on the project for at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers.
    * Is supportive of new and occasional contributors and helps get useful PRs in shape to merge
* Additional privileges:

>     * Has rights to approve pull requests in the Podman project or a subproject

Reviewer 2: I don't think it is defined what approving means? It is not clear to me what it means, is it simply to say it has been reviewed by a reviewer?

Author: Github's "approve" functionality, and/or the `/approve` function in the bot

#### The process of becoming a Reviewer is:
1. The contributor must be sponsored by a Maintainer. That sponsor will open a PR against the appropriate repository, which adds the nominee to the MAINTAINERS.md file as a reviewer.
2. The contributor will add a comment to the pull request indicating their willingness to assume the responsibilities of a Reviewer.
3. At least two Maintainers of the repository must concur to merge the PR.

### Maintainer
Description: Maintainers are established contributors with deep technical knowledge of the Podman project and/or one of its subprojects. Maintainers are granted the authority to merge pull requests, and are expected to participate in making decisions about the strategy and priorities of the project. Maintainers are responsible for code review and merging in a single repository or subproject. It is possible to become Maintainer of additional repositories or subprojects, but each additional repository or project will require a separate application and vote. They are able to participate in all maintainer activities, including Core Maintainer meetings, but do not have a vote at Core Maintainer meetings.

> In repositories using an OWNERS file, Maintainers are listed as Approvers in that file.

Reviewer: If this is not a common file to be shared, I'd include links to things like the OWNERS file.  If it will be common to more than one project, ignore this.

Author: This will be referred to by Buildah, Podman, and Skopeo repos.

In repositories using an OWNERS file, Maintainers are listed as Approvers in that file.

A Maintainer must meet the responsibilities and requirements of a Reviewer, plus:
* Responsibilities include:

>     * Sustained high level of reviews of pull requests to the project or subproject, with a goal of one or more a week when averaged across the year.

Reviewer 2: looks like cncf added us to their stat tracking, I have not looked into how they count PR reviews but [here is the filter](https://podmancontainertools.devstats.cncf.io/d/46/pr-reviews-by-contributor?orgId=1&var-period=w&var-repo_name=All&var-reviewers=%22Luap99%22&var-reviewers=%22mtrmac%22&var-reviewers=%22nalind%22&var-reviewers=%22baude%22&var-reviewers=%22giuseppe%22&var-reviewers=%22mheon%22&var-reviewers=%22rhatdan%22&var-reviewers=%22lsm5%22&var-reviewers=%22TomSweeneyRedHat%22&var-reviewers=%22cevich%22&var-reviewers=%22Honny1%22&var-reviewers=%22l0rd%22&var-reviewers=%22inknos%22&var-reviewers=%22flouthoc%22&var-reviewers=%22ygalblum%22&var-reviewers=%22vrothberg%22&var-reviewers=%22cgwalters%22&var-reviewers=%22jakecorrenti%22&var-reviewers=%22ashley-cui%22&var-reviewers=%22gtjoseph%22&var-reviewers=%22jwhonce%22&var-reviewers=%22arsenalzp%22&var-reviewers=%22meln5674%22&var-reviewers=%22kolyshkin%22&var-reviewers=%22dashea%22&var-reviewers=%22QiWang19%22&var-reviewers=%22kwilczynski%22&var-reviewers=%22vmasutin%22&var-reviewers=%22martinpitt%22&var-reviewers=%22jnovy%22&var-reviewers=%22cleverhu%22&var-reviewers=%22Jamstah%22&var-reviewers=%22dfr%22&var-reviewers=%22sbrivio-rh%22&var-reviewers=%22eriksjolund%22&var-reviewers=%22hanwen-flow%22&var-reviewers=%22mohanboddu%22&var-reviewers=%22axel7083%22&var-reviewers=%22Fak3%22&var-reviewers=%22Odilhao%22&var-reviewers=%22lpcalisi%22&var-reviewers=%22elezar%22&var-reviewers=%22kanlac%22&var-reviewers=%22lstocchi%22&var-reviewers=%22Neil-Smith%22&var-reviewers=%22eighthave%22&var-reviewers=%22tangentsoft%22&var-reviewers=%22cyphar%22&var-reviewers=%22silver886%22&var-reviewers=%22AhmedMoalla%22&var-reviewers=%22lslavkov%22&var-reviewers=%22hdub-tech%22&var-reviewers=%22apollo13%22&var-reviewers=%22saschagrunert%22&var-reviewers=%22arixmkii%22&var-reviewers=%22Jookia%22&var-reviewers=%22FedeDP%22&var-reviewers=%22M1cha%22&var-reviewers=%22edsantiago%22&var-reviewers=%22andrew-sayers%22&var-reviewers=%22aeijdenberg%22&var-reviewers=%22Gauravtalreja1%22&var-reviewers=%22mathstuf%22&var-reviewers=%22lazka%22&var-reviewers=%22lalfie-rg%22&var-reviewers=%22AlekseiNikiforovIBM%22&var-reviewers=%22brbayes-msft%22&var-reviewers=%22hswong3i%22&var-reviewers=%22ddarrah%22&var-reviewers=%22cyqsimon%22&var-reviewers=%22umohnani8%22&var-reviewers=%22bachp%22&var-reviewers=%22setharnold%22&var-reviewers=%22jesperronn%22&var-reviewers=%22de1987%22&from=now-1y&to=now) I came up with if we consider reviews over the last year per week, on the right it shows the averages for each person.

Nothing unusual in terms of names there, personally I think 1 is a pretty low bar but at the same time I am also not a fan of having "arbitrarily" interactions counts, because then people might start to game the metrics.

Reviewer: Confirm. I think one a week is a bit high for a maintainer; I'd go one a month. However, this is a reasonable level for a core maintainer.  I, too, dislike an arbitrary number

Reviewer: and re-reading, it's a goal, not a requirement, so not as bad.

    * Mentoring new Reviewers
    * Participating in CNCF maintainer activities for the projects they are maintainers of
    * Assisting Core Maintainers in determining strategy and policy for the project

>     * Participating in, and leading, community meetings

Author: They'll have to be "Podman Project" community meetings from here on out

>     * Participating in, and leading, community meetings

Author: The CNCF submission (which they are referring to as Podman Container Tools, but I've just designated the Podman project in this doc) is Podman, Buildah, and Skopeo, which will eventually be getting their own separate GH org (IE moving out of containers/ on Github) to make it very clear what is CNCF and what isn't, but before that happens we'll update the readmes on relevant projects to indicate they are/are not CNCF.

(As for the c/ libraries, Netavark/Aardvark - we can potentially add those to the Podman Project CNCF at a later date as subprojects. We'd like to get settled in and accustomed to running things in public before we expand further, though.)

    * Demonstrates a broad knowledge of all components, repositories, and subprojects of the Podman project.
    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
    * Mentors new Maintainers and Core Maintainers
    * Able to make decisions and contributions affecting the whole project, including multiple subprojects and repositories
    * Can commit to maintaining a high level of contribution to the project as a whole
* Additional privileges:
    * Represent the project in public as a senior project member
	* Represent the project in interactions with the CNCF
    * Have a vote in Core Maintainer decision-making meetings

#### Process of becoming a Core Maintainer:
1. A current maintainer must be sponsored by Core Maintainer. The Core Maintainer will open a PR against the main Podman repository and add the nominee as a Core Maintainer in the MAINTAINERS.md file.
2. The nominee will add a comment to the PR confirming that they agree to all requirements and responsibilities of becoming a Core Maintainer.
3. A majority of the current Core Maintainers must then approve the PR.

> 4. If, for some reason, all existing members are inactive according to the Inactivity policy below or there are no Core Maintainers due to resignations, a supermajority (66%) vote of maintainers can bypass this process and approve new Core Maintainers directly.

Reviewer: This section seems to be for Podman only, while the rest of the document seems to be for use by multiple projects.

Reviewer: I'll take that back, I see what's happening now.

Author: We need one repo to be the "main repo" for purposes of the project as a whole. I'm using Podman as that given that's what we were accepted under. Podman will have the definitive source of truth for multi-repo roles

* Responsibilities include:
    * Participating in CNCF maintainer activities
    * Arranging, participating in, and leading, community meetings
    * Managing the project website and gathering associated metrics
    * Managing the project’s social media accounts and mailing lists and gathering associated metrics
* Requirements
    * Sustained high level of contribution to the community, including attending and engaging in community meetings, contributions to the website, and contributions to documentation, for at least six months
    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
    * Can commit to maintaining a high level of contribution to the project's community, website, and social media presence
    * Advocates for the community in Maintainer and Core Maintainer meetings
* Additional privileges:
    * Represent the project in public

> 	* Represent the project in interactions with the CNCF

Reviewer: ditto extra ident here.

### Emeritus Maintainer
Emeritus Maintainers are former Maintainers or Core Maintainers whose status has lapsed, either voluntarily or through inactivity. We recognize that these former maintainers still have valuable experience and insights, and maintain Emeritus status as a way of recognizing this. Emeritus Maintainer also offers a fast-tracked path to becoming a Maintainer again, should the contributor wish to return to the project.

Emeritus Maintainers have no responsibilities or requirements beyond those of an ordinary Contributor.

#### Process of becoming an Emeritus Maintainer:
1. A current Maintainer or Core Maintainer may voluntarily resign from their position by making a pull request changing their role in the OWNERS file. They may choose to remove themselves entirely or to change their role to Emeritus Maintainer.
2. Maintainers and Core Maintainers removed due to the Inactivity policy below may be moved to Emeritus Status.

> ## Maintainers File

Reviewer 3: should this PR introduce that file?

Author: It should, yes. I'll do that tomorrow.
I'll also start up a reviewers guide for PRs.

Involuntary removal or demotion of Maintainers and Reviewers is handled through a vote by a majority of the current Maintainers. Core Maintainers may be involuntarily removed by a majority vote of current Core Maintainers or, if all Core Maintainers have stepped down or are inactive according to the inactivity policy, by a supermajority (66%) vote of maintainers.

## Stepping Down/Emeritus Process
If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).

Maintainers and Reviewers should contact the Maintainers about changing to Emeritus status, or reducing your contributor level. Core Maintainers should contact other Core Maintainers.

## Updates to this document
Updates to this Governance document require approval from a majority vote of the Core Maintainers.

## Contact
* For inquiries, please reach out to:

>     *  Tom Sweeney, Community Manager

Reviewer 2: might as well add the email address here so people don't have to look up contact info elsewhere

> Involuntary removal or demotion of Maintainers and Reviewers is handled through a vote by a majority of the current Maintainers. Core Maintainers may be involuntarily removed by a majority vote of current Core Maintainers or, if all Core Maintainers have stepped down or are inactive according to the inactivity policy, by a supermajority (66%) vote of maintainers.

Author: I think we should probably do complaints through a vote of the core maintainers, I don't know if I want to add a role with arbitrary remove-from-project permissions without a check

    * [Reviewer](#reviewer)
    * [Maintainer](#maintainer)
    * [Core Maintainer](#coremaintainer)
    * [Community Manager](#communitymanager)
* [Inactivity](#inactivity)
* [Involuntary Removal](#involuntary-removal-or-demotion)
* [Stepping Down/Emeritus Process](#stepping-downemeritus-process)
* [Updates to this Document](#updates-to-this-document)
* [Contact](#contact)

## Contributor Ladder

Hello! We are excited that you want to learn more about our project contributor ladder! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance as their involvement in the project grows.  Our project members are happy to help you advance along the contributor ladder.  At all levels, contributors are required to follow the CNCF Code of Conduct (COC).

Each of the project member roles below is organized into lists of three types of things.

* "Responsibilities" – functions of a member
* "Requirements" –  qualifications of a member
* "Privileges" –  entitlements of member

### Contributor

> Description: A Contributor contributes directly to the project and adds value to it. Contributions need not be code. People at the Contributor level may be new contributors, or they may only contribute occasionally.

Reviewer 4: ```suggestion
Description: A Contributor supports the project and adds value to it. Contributions need not be code. People at the Contributor level may be new contributors, or they may only contribute occasionally.
```
Pet peeve about stutter "Contributor contributes"

* Responsibilities include:
    * Follow the CNCF CoC
    * Follow the project contributing guide
* Requirements (one or several of the below):
    * Report and sometimes resolve issues against any of the project’s repositories
    * Occasionally submit PRs against any of the project’s repositories
    * Contribute to project documentation, including the manpages, tutorials, and Podman.io
    * Attend community meetings when reasonable
    * Answer questions from other community members on the mailing list, Slack, Matrix, and other communication channels

>     * Assist in triaging issues

Reviewer 4: ```suggestion
    * Assist in triaging issues, following the [triage guide](./TRIAGE.md). Adding comments to issues as appropriate. 
```

    * Assist in reviewing pull requests, including testing patches when applicable
    * Test release candidates and provide feedback
    * Promote the project in public
    * Help run the project infrastructure
* Privileges:
    * Invitations to contributor events
    * Eligible to become a Reviewer

### Reviewer
Description: A Reviewer has responsibility for the triage of issues and review of pull requests on the Podman project or a subproject, consisting of one or more of the Git repositories that form the project. They are collectively responsible, with other Reviewers, for reviewing changes to the repository or repositories and indicating whether those changes are ready to merge. They have a track record of contribution and review in the project.

Reviewers have all the rights and responsibilities of a Contributor, plus:

* Responsibilities include:
    * Regular contribution of pull requests to the Podman project or its subprojects
    * Triage of Github issues on the Podman project or its subprojects
    * Regularly fixing Github issues on the Podman project or its subprojects

>     * Following the [reviewing guide](./REVIEWING.md)

Reviewer 4: ```suggestion
    * Following the guides for [reviewing](./REVIEWING.md) and [triage](./TRIAGE.md)
```

    * A sustained high level of pull request reviews on the Podman project or one of its subprojects

>     * Assisting new Contributors with the process of contributing

Reviewer 4: ```suggestion
    * Assisting new Contributors with the process of supporting these projects
```

    * Helping other contributors become reviewers
* Requirements:

>     * Has a proven record of good-faith contributions to the project as a Contributor for a period of at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers and Core Maintainers.

Reviewer 4: ```suggestion
    * Has a proven record of good-faith contributions to the project as a Contributor for a period of at least 6 months. The time requirement may be overridden by a [super-majority](https://en.wikipedia.org/wiki/Supermajority#Two-thirds_vote)  vote of Maintainers and Core Maintainers.
```

    * Has participated in pull request review and/or issue triage on the project for at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers and Core Maintainers.
    * Is supportive of new and occasional contributors and helps get useful PRs in shape to merge
* Additional privileges:
    * Has rights to approve pull requests in the Podman project or a subproject, marking them as ready for a Maintainer to review and merge
    * Can recommend and review other contributors to become Reviewers

In repositories using an OWNERS file, Reviewers are listed as Reviewers in that file.

#### The process of becoming a Reviewer is:
1. The contributor must be sponsored by a Maintainer. That sponsor will open a PR against the appropriate repository, which adds the nominee to the MAINTAINERS.md file as a reviewer.
2. The contributor will add a comment to the pull request indicating their willingness to assume the responsibilities of a Reviewer.
3. At least two Maintainers of the repository must concur to merge the PR.

### Maintainer

> Description: Maintainers are very established contributors with deep technical knowledge of the Podman project and/or one of its subprojects. Maintainers are granted the authority to merge pull requests, and are expected to participate in making decisions about the strategy and priorities of the project. Maintainers are responsible for code review and merging in a single repository or subproject. It is possible to become Maintainer of additional repositories or subprojects, but each additional repository or project will require a separate application and vote. They are able to participate in all maintainer activities, including Core Maintainer meetings, but do not have a vote at Core Maintainer meetings.

Reviewer 4: ```suggestion
Description: Maintainers are established contributors with deep technical knowledge of the Podman project and/or one of its subprojects. Maintainers are granted the authority to merge pull requests, and are expected to participate in making decisions about the strategy and priorities of the project. Maintainers are responsible for code review and merging in a single repository or subproject. It is possible to become Maintainer of additional repositories or subprojects, but each additional repository or project will require a separate application and vote. They are able to participate in all maintainer activities, including Core Maintainer meetings, but do not have a vote at Core Maintainer meetings.
```
`very` is not measurable here.

2. The nominee will add a comment to the PR confirming that they agree to all requirements and responsibilities of becoming a Maintainer.
3. A majority of the current Maintainers of the repository or subproject (including Core Maintainers) must then approve the PR. The need for a majority is removed for Emeritus Maintainers, who require only 2 current Maintainers or Core Maintainers to approve their return.

### Core Maintainer
Description: As the Podman project is composed of a number of subprojects, most maintainers will not have full knowledge of the full project and all its technical aspects. Those that do are eligible to become Core Maintainers, responsible for decisions affecting the entire project. Core Maintainers may act as a maintainer in all repositories and subprojects of the Podman Project.

* Responsibilities include:
    * All responsibilities of a maintainer
    * Determining strategy and policy for the project

>     * Participating in, and leading, community meetings

Reviewer 4: ```suggestion
    * Participating in, and leading, community meetings
    * Communicate with the CNCF on behalf of the project
```

Author: Already says "All responsibilities of a Maintainer" which includes that. Though I should remove the "Participating in community meetings bit" as it's duplicated from the Maintainer list

## Inactivity

* Inactivity is measured by one or more of the following:
    * Periods of no contribution of code, pull request review, or participation in issue triage for longer than 12 months
    * Periods of no communication for longer than 3 months
* Consequences of being inactive include:
    * Involuntary removal or demotion
    * Being asked to move to Emeritus status

## Involuntary Removal or Demotion

Involuntary removal/demotion of a contributor happens when responsibilities and requirements aren't being met. This may include repeated patterns of inactivity, an extended period of inactivity, a period of failing to meet the requirements of your role, and/or a violation of the Code of Conduct. This process is important because it protects the community and its deliverables while also opening up opportunities for new contributors to step in.

> Involuntary removal or demotion of Maintainers and Reviewers is handled through a vote by a majority of the current Maintainers. Core Maintainers may be involuntarily removed by a majority vote of current Core Maintainers or, if all Core Maintainers have stepped down or are inactive according to the inactivity policy, by a supermajority (66%) vote of maintainers.

Reviewer 4: ```suggestion
Involuntary removal or demotion of Maintainers and Reviewers is handled through a vote by a majority of the current Maintainers. Core Maintainers may be involuntarily removed by a majority vote of current Core Maintainers or, if all Core Maintainers have stepped down or are inactive according to the inactivity policy, by a [super-majority](https://en.wikipedia.org/wiki/Supermajority#Two-thirds_vote) vote of maintainers.
```

## Stepping Down/Emeritus Process
If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).

Maintainers and Reviewers should contact the Maintainers about changing to Emeritus status, or reducing your contributor level. Core Maintainers should contact other Core Maintainers.

## Updates to this document

> Updates to this Governance document require approval from a majority vote of the Core Maintainers.

Reviewer 4: ```suggestion
Updates to this Governance document require approval from a majority vote of the Core Maintainers.
```
Should this be a majority or super-majority?

Author: I'll change it. Conveniently, with our current 6, majority and supermajority are identical.

# Contributor Ladder Template

* [Contributor Ladder](#contributor-ladder-template)
    * [Contributor](#contributor)
    * [Reviewer](#reviewer)
    * [Maintainer](#maintainer)
    * [Core Maintainer](#coremaintainer)
    * [Community Manager](#communitymanager)
* [Inactivity](#inactivity)
* [Involuntary Removal](#involuntary-removal-or-demotion)
* [Stepping Down/Emeritus Process](#stepping-downemeritus-process)
* [Updates to this Document](#updates-to-this-document)
* [Contact](#contact)

## Contributor Ladder

> Hello! We are excited that you want to learn more about our project contributor ladder! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance up it as their involvement in the project grows.  Our project members are happy to help you advance along the contributor ladder.  At all levels, contributors are required to follow the CNCF Code of Conduct (COC).

Reviewer 5: ```suggestion
Hello! We are excited that you want to learn more about our project contributor ladder! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance as their involvement in the project grows.  Our project members are happy to help you advance along the contributor ladder.  At all levels, contributors are required to follow the CNCF Code of Conduct (COC).
```

    * Occasionally submit PRs against any of the project’s repositories
    * Contribute to project documentation, including the manpages, tutorials, and Podman.io
    * Attend community meetings when reasonable
    * Answer questions from other community members on the mailing list, Slack, Matrix, and other communication channels
    * Assist in triaging issues
    * Assist in reviewing pull requests, including testing patches when applicable
    * Test release candidates and provide feedback
    * Promote the project in public
    * Help run the project infrastructure
* Privileges:
    * Invitations to contributor events
    * Eligible to become a Reviewer

### Reviewer
Description: A Reviewer has responsibility for the triage of issues and review of pull requests on the Podman project or a subproject, consisting of one or more of the Git repositories that form the project. They are collectively responsible, with other Reviewers, for reviewing changes to the repository or repositories and indicating whether those changes are ready to merge. They have a track record of contribution and review in the project.

> Reviewers have all the rights and responsibilities of an Contributor, plus:

Reviewer 5: ```suggestion
Reviewers have all the rights and responsibilities of a Contributor, plus:
```

* Responsibilities include:
    * Regular contribution of pull requests to the Podman project or its subprojects
    * Triage of Github issues on the Podman project or its subprojects
    * Regularly fixing Github issues on the Podman project or its subprojects

>     * Following the reviewing guide

Reviewer 5: Can we link to the guideline if there is one?

Author: Done

    * Has a proven record of good-faith contributions to the project as a Contributor for a period of at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers.
    * Has participated in pull request review and/or issue triage on the project for at least 6 months. The time requirement may be overridden by a supermajority vote of Maintainers.
    * Is supportive of new and occasional contributors and helps get useful PRs in shape to merge
* Additional privileges:
    * Has rights to approve pull requests in the Podman project or a subproject, marking them as ready for a Maintainer to review and merge
    * Can recommend and review other contributors to become Reviewers

All Reviewers are listed in the OWNERS file of the repository or repositories they have review authority over.

#### The process of becoming a Reviewer is:
1. The contributor must be sponsored by a Maintainer. That sponsor will open a PR against the appropriate repository, which adds the contributor to the MAINTAINERS.md file as a reviewer.
2. The contributor will add a comment to the pull request indicating their willingness to assume the responsibilities of a Reviewer.

> 3. At least two members of the team that owns the repository in question, who are already Maintainers, must concur to merge the PR.

Reviewer 5: ```suggestion
3. At least two team members that own the repository in question, who are already Maintainers, must concur to merge the PR.
```

### Maintainer

> Description: Maintainers are very established contributors with deep technical knowledge of the Podman project and/or one of its subprojects. Maintainers are granted the authority to merge pull requests, and are expected to participate in making decisions about the strategy and priorities of the project. Maintainers are responsible for code review and merging in a single repository or subproject. It is possible to become Maintainer of additional repositories or subprojects, but each additional repository or project will require a separate application and vote They are able to participate in all maintainer activities, including Core Maintainer meetings, but do not have a vote at Core Maintainer meetings.

Reviewer 5: ```suggestion
Description: Maintainers are very established contributors with deep technical knowledge of the Podman project and/or one of its subprojects. Maintainers are granted the authority to merge pull requests, and are expected to participate in making decisions about the strategy and priorities of the project. Maintainers are responsible for code review and merging in a single repository or subproject. It is possible to become Maintainer of additional repositories or subprojects, but each additional repository or project will require a separate application and vote. They are able to participate in all maintainer activities, including Core Maintainer meetings, but do not have a vote at Core Maintainer meetings.
```

    * Merging pull requests which pass review
    * Mentoring new Reviewers
    * Participating in CNCF maintainer activities for the projects they are maintainers of
    * Assisting Core Maintainers in determining strategy and policy for the project
    * Participating in, and leading, community meetings
* Requirements
    * Experience as a Reviewer for at least 6 months, or status as an Emeritus Maintainer. The time requirement may be overridden by a supermajority vote of Maintainers.
    * Demonstrates a broad knowledge of the project or one or more of its subprojects
    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
    * Mentors contributors, reviewers, and new maintainers
    * Collaborates with other Maintainers to work on complex contributions
    * Can commit to maintaining a high level of contribution to the project or one of its subprojects
* Additional privileges:
    * Represent the project in public as a senior project member
    * Communicate with the CNCF on behalf of the project

> 	* Have a voice, but not a vote, in Core Maintainer decision-making meetings

Reviewer 5: ```suggestion
    * Have a voice, but not a vote, in Core Maintainer decision-making meetings
```

* Responsibilities include:
    * Participating in CNCF maintainer activities
    * Arranging, participating in, and leading, community meetings
    * Managing the project website and gathering associated metrics
    * Managing the project’s social media accounts and mailing lists and gathering associated metrics
* Requirements
    * Sustained high level of contribution to the community, including attending and engaging in community meetings, contributions to the website, and contributions to documentation, for at least six months
    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
    * Can commit to maintaining a high level of contribution to the project's community, website, and social media presence
    * Advocates for the community in Maintainer and Core Maintainer meetings
* Additional privileges:
    * Represent the project in public

>     * Communicate with the CNCF on behalf of the project

Reviewer 6: Do you want just "Communicate" with the CNCF or do you want the ability to represent on behalf of the project? 

For example do you want this person to be able to ask for resources, do event coordination with us, submit maintainer CFPs, be able to file support tickets on behalf of other maintainers, access to #maintainers-circle, etc. 

If so I'd strengthen this a tad so we know to give them the proper rights.

Author: I was aiming for the same rights as maintainers, so I'll use the same wording we use for them. Good catch.

> 	* Communicate with the CNCF on behalf of the project
> 	* Have a voice, but not a vote, in Core Maintainer decision-making meetings

Reviewer 5: ```suggestion
    * Communicate with the CNCF on behalf of the project
    * Have a voice, but not a vote, in Core Maintainer decision-making meetings
```

    * [Emeritus Maintainer](#emeritus-maintainer)
* [Maintainers File](#maintainers-file)
* [Inactivity](#inactivity)
* [Involuntary Removal](#involuntary-removal-or-demotion)
* [Stepping Down/Emeritus Process](#stepping-downemeritus-process)
* [Updates to this Document](#updates-to-this-document)
* [Contact](#contact)

# Podman Project

This document defines the governance of the Podman Project, including its subprojects. It defines the various roles our maintainers fill, how to become a maintainer, and how project-level decisions are made.

# Contributor Ladder

The Podman project has a number of maintainer roles arranged in a ladder. Each role is a rung on the ladder, with different responsibilities and privileges. Community members generally start at the first levels of the "ladder" and advance as their involvement in the project grows. Our project members are happy to help you advance along the contributor ladder. At all levels, contributors are required to follow the CNCF Code of Conduct (COC).

Each of the project member roles below is organized into lists of three types of things.

* "Responsibilities" – functions of a member
* "Requirements" –  qualifications of a member

> * "Privileges" –  entitlements of member

Reviewer 7: Non-blocking, non-substantive:

Consider moving this to a comment. Reading this document, I emotionally feel that starting with a “terminology” section, the way some laws do, says “this was written by a bureaucracy for a bureaucracy, expect the processes to be painfully formalistic”.

I do agree that having some common structure to the ladder entries is useful, but maybe that structure can be understood well enough from context when reading the actual entries.

[I’m even wondering about not describing the ladder at all, but that’s really just a question, not even a weak preference.]

---

(This is exacerbated in that I’m also wondering what “responsibilities” are; starting with a terminology section _I don’t immediately understand_ worries me. The “responsibilities” are not requirements / expectations, that’s a different section. But then again, the first example says “follow the CNCF CoC”, which is probably a hard requirement. And in the other ladder entries, they are closer to “privileges”. Is it that “requirements are how we quantify responsibilities”? Are “responsibilities” closer to ”typical kinds of contribution”? OTOH sometimes they imply specific access rights on the project.

_Maybe_ the “follow CoC”, part which is already in the paragraph above, can be removed from the “contributor” ladder entry, and then `s/Responsibilities/Typical activities/`?

Ultimately all of this matters very little.)

    * Regularly fixing GitHub issues on the Podman project or its subprojects
    * Following the [reviewing guide](./REVIEWING.md) and [issue triage guide](./TRIAGE.md)
    * A sustained high level of pull request reviews on the Podman project or one of its subprojects
    * Assisting new Contributors in their interactions with the project
    * Helping other contributors become reviewers
* Requirements:
    * Has a proven record of good-faith contributions to the project as a Contributor for a period of at least 6 months. The time requirement may be overridden by a supermajority (66%) vote of Maintainers and Core Maintainers.
    * Has participated in pull request review and/or issue triage on the project for at least 6 months. The time requirement may be overridden by a supermajority (66%) vote of Maintainers and Core Maintainers.
    * Is supportive of new and occasional contributors and helps get useful PRs in shape to merge
* Additional privileges:
    * Has rights to approve pull requests in the Podman project or a subproject, marking them as ready for a Maintainer to review and merge
    * Can recommend and review other contributors to become Reviewers

> 	* Has permissions to change labels on Github to aid in triage

Reviewer 8: ```suggestion
    * Has permissions to change labels on Github to aid in triage
```

    * Sustained high level of reviews of pull requests to the project or subproject, with a goal of one or more a week when averaged across the year.
    * Merging pull requests which pass review
    * Mentoring new Reviewers
    * Participating in CNCF maintainer activities for the projects they are maintainers of
    * Assisting Core Maintainers in determining strategy and policy for the project
    * Participating in, and leading, community meetings
* Requirements
    * Experience as a Reviewer for at least 6 months, or status as an Emeritus Maintainer. The time requirement may be overridden by a supermajority (66%) vote of Maintainers and Core Maintainers.
    * Demonstrates a broad knowledge of the project or one or more of its subprojects
    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
    * Mentors contributors, reviewers, and new maintainers
    * Collaborates with other Maintainers to work on complex contributions
    * Can commit to maintaining a high level of contribution to the project or one of its subprojects
* Additional privileges:
    * Represent the project in public as a senior project member

> 	* Represent the project in interactions with the CNCF

Reviewer 8: ```suggestion
    * Represent the project in interactions with the CNCF
```
